framework/helpers/BaseHtml.php

```diff
@@ -1981,6 +1981,11 @@ public static function renderTagAttributes($attributes)
                     if (empty($value)) {
                         continue;
                     }
+                    if (count($value) > 1) {
+                        // removes duplicate classes
+                        $value = explode(' ', implode(' ', $value));
```

Comment:
That would potentially slow down existing code where classes are specified as arrays. Not sure it worth it since having some duplicates doesn't really break anything.

Reply:
@samdark
> doesn't really break anything.

it's not true, for example:
~~~css
.c { font-size: 1rem}
.c2 {font-size: 2rem}
.c3 {font-weight: bold}
~~~
merge `'c c3'` to `['widget' => 'c c2']`:
expected result: `class="c  c2 c3"` - `font-size: 2rem`
actual result: `class="c  c2 c c3"` - `font-size: 1rem`

Reply:
Yes, that is valid case.